src/dependencies.js

```diff
@@ -84,25 +84,38 @@ const getFileDependencies = async function({ path, packageJson, state }) {
   const dependencies = precinct.paperwork(path, { includeCore: false })
 
   const depsPaths = await Promise.all(
-    dependencies.map(dependency => getImportDependencies({ dependency, basedir, packageJson, state }))
+    dependencies.map(dependency => getImportDependencies({ dependency, basedir, packageJson, state, treeShakeNext }))
   )
   return [].concat(...depsPaths)
 }
 
-const getImportDependencies = function({ dependency, basedir, packageJson, state }) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
-    return getTreeShakedDependencies({ dependency, basedir, packageJson, state })
+const getImportDependencies = function({ dependency, basedir, packageJson, state, treeShakeNext }) {
+  const shouldTreeShakeNext = treeShakeNext || isNextOnNetlify(dependency)
+  if (shouldTreeShake(dependency, shouldTreeShakeNext)) {
+    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakeNext: shouldTreeShakeNext })
   }
 
   return getAllDependencies({ dependency, basedir, state, packageJson })
 }
 
+const isNextOnNetlify = function(dependency) {
+  return dependency === './renderNextPage'
+}
+
+const shouldTreeShake = function(dependency, treeShakeNext) {
```

Comment:
We might want to add some code comment to explain why we are doing this?

Reply:
Let me know if this https://github.com/netlify/zip-it-and-ship-it/pull/242/commits/5ed3906dbf853073c0ed4a393935efc7f76a3caf is clear enought

Reply:
`since there is no guarantee that would work for every module`: maybe we should explain that this is because some Node.js modules either: 1. do dynamic requires or 2. read files, which would not work with our basic tree shaking logic.

Apart from this, looks good :+1:

Reply:
Done in https://github.com/netlify/zip-it-and-ship-it/pull/242/commits/6f4a9e0912131ed85d22f0dabd3d528a0d13cefd